core/src/main/java/io/aiven/kafka/tieredstorage/transform/RateLimitedInputStream.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2024 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.tieredstorage.transform;
+
+import java.io.FilterInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.time.Duration;
+
+import io.github.bucket4j.Bucket;
+
+/**
+ * Input Stream with a limited rate of reading from source.
+ * Aims to provide a stable upload speed that does not affect other streams of work in a broker.
+ * Rate limiting is only implemented for {@code InputStream#read(bytes, offset, len)} as rate limiting individual reads
+ * is too expensive.
+ */
+public class RateLimitedInputStream extends FilterInputStream {
+    static final int DEFAULT_BUFFER_SIZE = 8192;
```

Comment:
This constant servers like `MIN_RATE` or something like this. Maybe it should be renamed? Or if I misunderstood its purpose, maybe a comment?

Reply:
Sure, adding comments and renaming to min_rate